includes/classes/shopping_cart.php

```diff
@@ -1212,181 +1259,197 @@ public function get_products($check_for_valid_cart = false)
-                                    $this->remove($products_id);
-                                    break;
-                                }
-                            }
-                        }
-                    }
+            if ($check_for_valid_cart == true) {
+                if (empty($this->flag_duplicate_quantity_msgs_set['keep'])) {
+                    $this->flag_duplicate_quantity_msgs_set = [];
+                }
+                $fix_once = 0;
+                // Check products_status if not already
+                if ($product['products_status'] === '0') {
+                    $fix_once++;
+                    $_SESSION['valid_to_checkout'] = false;
+                    $_SESSION['cart_errors'] .= ERROR_PRODUCT . $product['products_name'] . ERROR_PRODUCT_STATUS_SHOPPING_CART . '<br>';
+                    $this->remove($uprid);
+                    continue;
+                }
 
-                    // check only if valid products_status
-                    if ($fix_once == 0) {
-                        $check_quantity = $this->contents[$products_id]['qty'];
-                        $check_quantity_min = $products->fields['products_quantity_order_min'];
-                        // Check quantity min
-                        if ($new_check_quantity = $this->in_cart_mixed($prid)) {
-                            $check_quantity = $new_check_quantity;
-                        }
-                    }
+                if (isset($data['attributes'])) {
+                    foreach ($data['attributes'] as $value) {
+                        $sql = "SELECT products_id
+                                FROM " . TABLE_PRODUCTS_ATTRIBUTES . " pa
+                                WHERE pa.products_id = $prid
+                                AND pa.options_values_id = " . (int)$value;
 
-                    // Check Quantity Max if not already an error on Minimum
-                    if ($fix_once == 0) {
-                        if ($products->fields['products_quantity_order_max'] != 0 && $check_quantity > $products->fields['products_quantity_order_max'] && !isset($this->flag_duplicate_quantity_msgs_set[(int)$prid]['max'])) {
+                        $chk_attributes_exist = $db->Execute($sql);
+                        if ($chk_attributes_exist->EOF) {
                             $fix_once++;
                             $_SESSION['valid_to_checkout'] = false;
-                            $_SESSION['cart_errors'] .= ERROR_PRODUCT . $products->fields['products_name'] . ERROR_PRODUCT_QUANTITY_MAX_SHOPPING_CART . ERROR_PRODUCT_QUANTITY_ORDERED . $check_quantity . ' <span class="alertBlack">' . zen_get_products_quantity_min_units_display((int)$prid, false, true) . '</span> ' . '<br>';
-                            $this->flag_duplicate_quantity_msgs_set[(int)$prid]['max'] = true;
+                            $chk_products_link =
```

Comment:
Q: This removed the check for `products_status`, and now renders a link regardless of status, instead of only displaying the name if the status is 0 (product disabled).
Was this intended?

Reply:
This section isn't run if the `products_status` is 0; bypassed by the processing on lines 1314-1320 above.